rust_src/src/indent.rs

```diff
@@ -178,4 +180,126 @@ pub fn indent_to(column: EmacsInt, minimum: Option<EmacsInt>) -> EmacsInt {
+/// POS is the buffer position where the scan stopped.
+/// VPOS is the vertical position where the scan stopped.
+/// HPOS is the horizontal position where the scan stopped.
+///
+/// PREVHPOS is the horizontal position one character back from POS.
+/// CONTIN is t if a line was continued after (or within) the previous character.
+///
+/// For example, to find the buffer position of column COL of line
+/// LINE of a certain window, pass the window's starting location as
+/// FROM and the window's upper-left coordinates as FROMPOS.  Pass the
+/// buffer's (point-max) as TO, to limit the scan to the end of the
+/// visible section of the buffer, and pass LINE and COL as TOPOS.
+#[lisp_fn]
+pub fn compute_motion(
+    from: LispNumber,
+    frompos: LispCons,
+    to: LispNumber,
+    topos: Option<LispCons>,
+    width: Option<EmacsInt>,
+    offsets: Option<LispCons>,
+    window: LispWindowLiveOrSelected,
+) -> LispObject {
+    fn number_pair(cons: LispCons) -> [EmacsInt; 2] {
+        [cons.car().into(), cons.cdr().into()]
+    }
+    let [from_hpos, from_vpos] = number_pair(frompos);
+
+    let mut win: LispWindowRef = window.into();
+    let window_width = unsafe { window_body_width(win.as_mut(), false) } as EmacsInt
+        - if cfg!(feature = "window-system") && win.get_frame().is_gui_window() {
+            0
+        } else {
+            1
+        };
+
+    let arg_width = width.unwrap_or(-1);
+
+    let [to_hpos, to_vpos] = match topos {
+        None => [
+            window_width,
+            unsafe { window_internal_height(win.as_mut()) } as EmacsInt,
+        ],
+        Some(cons) => number_pair(cons),
+    };
+    let [hscroll, tab_offset] = match offsets {
+        None => [0, 0],
+        Some(cons) => number_pair(cons),
+    };
+    if !(0 <= hscroll
+        && hscroll <= isize::max_value() as EmacsInt
+        && 0 <= tab_offset
+        && tab_offset <= libc::INT_MAX as EmacsInt)
+    {
+        args_out_of_range!(hscroll, tab_offset);
+    }
+
+    let buffer = &mut ThreadState::current_buffer_unchecked();
+    let begv = buffer.begv as EmacsInt;
+    let zv = buffer.zv as EmacsInt;
+    if from.to_fixnum() < begv || from.to_fixnum() > zv {
+        args_out_of_range!(from, begv, zv);
+    }
+    if to.to_fixnum() < begv || to.to_fixnum() > zv {
+        args_out_of_range!(to, begv, zv);
+    }
+
+    let pos = unsafe {
+        *remacs_sys::compute_motion(
+            from.to_fixnum() as isize,
+            buf_charpos_to_bytepos(buffer.as_mut(), from.to_fixnum() as isize),
+            from_vpos,
+            from_hpos,
+            false,
+            to.to_fixnum() as isize,
+            to_vpos,
+            to_hpos,
+            arg_width,
+            hscroll as isize,
+            tab_offset as i32,
+            win.as_mut(),
+        )
+    };
+
+    list!(
+        pos.bufpos,
+        pos.hpos,
+        pos.vpos,
+        pos.prevhpos,
+        if pos.contin != 0 { Qt } else { Qnil }
```

Comment:
Or `(pos.contin != 0).into()`. Both forms are valid.

Reply:
turns out it even works without `into()`, because `cons` does that :)